includes/utilities.inc

```diff
@@ -104,12 +104,29 @@ function islandora_rest_accessible($endpoint, $request_method, $resource = FALSE
   if ($endpoint == 'solr') {
     return islandora_rest_solr_access();
   }
+
+  // Retrieve both islandora core and islandora rest based permissions.
   $perm = islandora_rest_resource_access_permission($endpoint, $request_method);
+  $access = array();
   if (is_subclass_of($resource, 'AbstractObject')) {
-    return islandora_object_access($perm, $resource);
+    // Call core permission check, to ensure XACML permissions are
+    // Respected.
+    array_push($access, islandora_object_access($perm['core'], $resource));
+    array_push($access, islandora_object_access($perm['rest'], $resource));
+
+    // Deny access if either permission returns no access.
+    return in_array(FALSE, $access) ? FALSE : TRUE;
   }
   elseif (is_subclass_of($resource, 'AbstractDatastream')) {
-    return islandora_datastream_access($perm, $resource);
+    // Only for use with islandora tokened object access permission.
+    if (is_callable($perm['core'])) {
+      array_push($access, islandora_object_datastream_tokened_access_callback(ISLANDORA_VIEW_OBJECTS, $resource->parent->id, $resource));
+    }
+    else {
+      array_push($access, islandora_datastream_access($perm['core'], $resource));
+    }
+    array_push($access, islandora_datastream_access($perm['rest'], $resource));
+    return in_array(FALSE, $access) ? FALSE : TRUE;
   }
   return user_access($perm);
```

Comment:
This would be broken by the [change of the return from `islandora_rest_resource_access_permission()`](https://github.com/discoverygarden/islandora_rest/pull/18/files#diff-2d69368f59ba81d5635cd8274dd66090R168); assumes the string here, but not it's an array.